src/common/libkvs/treeobj.h

```diff
@@ -69,12 +69,14 @@ json_t *treeobj_get_entry (json_t *obj, const char *name);
 int treeobj_insert_entry (json_t *obj, const char *name, json_t *obj2);
 int treeobj_delete_entry (json_t *obj, const char *name);
 
-/* Copy a treeobj dir.
- * This function performs a shallow copy on a dir object, copying the
- * first level of directory entries.  Use this function over json_copy()
- * on dir objects, as the latter is not safe on treeobj objects.
+/* Copy a treeobj.
+ * This function performs a shallow copy on a dir, dirref, or valref
+ * object.  Use this function over json_copy() on dir, dirref, or
+ * valref objects, as the latter is not safe on treeobj objects.
+ * This function will copy one additional level deeper in the treeobj
+ * so it is safe to manipulate.
  */
-json_t *treeobj_copy_dir (json_t *obj);
+json_t *treeobj_copy (json_t *obj);
```

Comment:
Should this be named so that it's more clear that it makes a thin copy of the `data` object?

This property is critical to understanding how commits are processed.

Maybe `treeobj_copy_thin()` or similar?

Reply:
Actually it is a deep copy as long as it's not a directory that includes "rolled up" subdirectories.  Hmm, maybe just need to add a new `treeobj_copy()` that makes a deep copy and leave that other one alone?

Reply:
I can see your point that ```treeobj_copy()``` by itself may be ambiguous.  I had wanted to avoid it, perhaps adding separate ```treeobj_copy_dirref()``` and ```treeobj_copy_valref()``` functions would be better?

Reply:
How about `treeobj_copy()` that fully copies any treeobj, and then rename `treeobj_copy_dir()` to something more indicitave of what it's doing like `treeobj_copy_dir_thin()`?  (the rename could be deferred to another PR)

Reply:
Now that I think about it, how about just making a function ```treeobj_copy_shallow()```.  It can be made quite clear that it's a shallow copy of the data within a treeobj, regardless of the data it is.